store/src/main/java/com/zimbra/cs/service/admin/IssueCert.java

```diff
@@ -0,0 +1,114 @@
+package com.zimbra.cs.service.admin;
+
+import com.zimbra.common.account.Key.DomainBy;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.soap.MailConstants;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Domain;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.account.Server;
+import com.zimbra.cs.account.accesscontrol.generated.AdminRights;
+import com.zimbra.cs.rmgmt.RemoteCertbot;
+import com.zimbra.cs.rmgmt.RemoteCommands;
+import com.zimbra.cs.rmgmt.RemoteManager;
+import com.zimbra.soap.ZimbraSoapContext;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Admin Handler class to issue a LetsEncrypt certificate for a domain using
+ * {@link com.zimbra.cs.rmgmt.RemoteManager}, {@link RemoteCertbot}.
+ *
+ * @author Yuliya Aheeva
+ * @since 23.3.0
+ */
+public class IssueCert extends AdminDocumentHandler {
+
+  /**
+   * Handles the request. Searches a domain by id, checks admin rights (accessible to global and
+   * delegated admin of requested domain), searches a server with proxy node, executes remote
+   * certbot command on it, creates response element.
+   *
+   * @param request {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertRequest}
+   * @param context request context
+   * @return {@link Element} representation of {@link
+   *     com.zimbra.soap.admin.message.IssueCertResponse}
+   * @throws ServiceException in case if domain could not be found, domain doesn't have
+   * PublicServiceHostname and at least one VirtualHostName,
+   * server with proxy node could not be found or domain admin dosn't have rights to deal with
+   * this domain.
+   *
+   * It won't throw an exception in case if remote execution command fails, instead will create
+   * a response and add a failure message to it.
+   */
+  @Override
+  public Element handle(final Element request, final Map<String, Object> context)
+      throws ServiceException {
+
+    ZimbraSoapContext zsc = getZimbraSoapContext(context);
+
+    Provisioning prov = Provisioning.getInstance();
+    String domainId = request.getAttribute(AdminConstants.A_DOMAIN);
+    Domain domain = prov.get(DomainBy.id, domainId);
+
+    if (domain == null) {
+      throw ServiceException.INVALID_REQUEST(
+          "Domain with id " + domainId + " could not be found.", null);
+    }
```

Comment:
Let's make our codebase more 2023 java style yo :sunglasses: 
```suggestion
    final Domain domain = Optional.ofNullable(prov.get(DomainBy.id, domainId)).orElseThrow(() -> ServiceException.INVALID_REQUEST(
          "Domain with id " + domainId + " could not be found.", null));
```

Less cleaner code :wink: